IUPAC_SMILES+.asciidoc

```diff
@@ -212,23 +227,21 @@ Examples:
 | `[ClH1]`  | hydrochloric acid     |
 |=================================================
 
-A hydrogen atom cannot have a hydrogen count, for example, `[HH1]` is illegal.  Hydrogens connected
+A hydrogen atom cannot have a hydrogen count, for example, `[HH1]` is invalid.  Hydrogens connected
 to other hydrogens must be represented as explicit atoms in square brackets. For example molecular
 hydrogen must be written as `[H][H]`.
 
-_Question: are more than 9 hydrogens possible?  Should they be
-supported?_
-
 [[charge]]
 
 ==== Charge
 
-Charge is specified by a `+n` or `-n` where `n` is a number; if the
-number is missing, it means either `+1` or `-1` as appropriate.
+Charge is specified by a `+n` or `-n` where `n` is a number from `0` to `15`. If a charge is specified without
+a number, a `1` is implied. If the number is `0`, the charge is interpreted as no charge. 
+That is, `[Ag+0]` and `[Ag-0]` are equivalent to `[Ag]`. A `0` charge should be avoided and is not recommended 
+as a best practice. A charge specification leading with `0` and followed be a second digit such as `[Ag+01]` is invalid.   
 
-For backwards compatibility, a general-purpose SMILES parser should
-accept the symbols `--` and `++` to mean charges of `-2` and `+2`, but this
-is a deprecated form and should be avoided.
+Repeated symbols such as `'--'` and `'++'` are valid and interpreted as charges of `-2` and `+2`, respectively. Symbols can
```

Comment:
Repeats of more than two symbols are not covered by the grammar, thus statement "Symbols can repeat up to 15 times" seems to contradict the grammar. As allowing more repeats is intentional (described in the changelog), grammar should be updated to reflect it.